### PR TITLE
Add frontend config and associate ssl policy

### PR DESCRIPTION
**Pull request by matthew-robinson-ons**
# What and why?
This PR introduces a frontend config to responses-dashboard and associates it with the ssl policy created in the terraform update. 
# How to test?
Not the easiest to test due to lack of loadbalancer outside pre-prod and PROD. Steps to test through dev environment:
- Deleted existing deployment and service from environment
- Created temporary ip address `gcloud compute addresses create dashboard-ip --global` (deleted after testing).
- Manually created [terraform](https://github.com/ONSdigital/ras-rm-terraform/pull/355/files) equivalent SSL policy associated with the ingress. 
- Use helm to deploy the pr version of responses-dashboard
          helm upgrade --install responses-dashboard _infra/helm/responses-dashboard --namespace "" \
           --set-string gcp.project="ras-rm-dev" \
           --set-string env="dev" \
           --set-string namespace="" \
           --set-string image.tag="pr-151"
- Cloud armor policy is required for the ingress, manually created basic deny rule through ui (deleted after testing).

The updated policy should then be associated to the frontend ingress.

# Jira
https://jira.ons.gov.uk/browse/RAS-1064

### CODE REVIEW

File: _infra/helm/responses-dashboard/values.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 env: dev
 namespace: dev
-publicIP: false
+publicIP: true
```
Comment (matthew-robinson-ons) [inline _infra/helm/responses-dashboard/values.yaml]: To revert before merge

File: _infra/helm/responses-dashboard/values.yaml [hunk 2]
```diff
@@ -10,7 +10,7 @@ image:
 container:
   port: 8080
 service:
-  type: ClusterIP
+  type: NodePort
```
Comment (matthew-robinson-ons) [inline _infra/helm/responses-dashboard/values.yaml]: To revert before merge

File: _infra/helm/responses-dashboard/values.yaml [hunk 3]
```diff
@@ -36,9 +36,14 @@ reportRefreshSeconds: 10
 loggingLevel: INFO
 
 ingress:
-  enabled: false
-  host: responses-dashboard.example.com
+  enabled: true
+  host: responses-dashboard.robinm3.example.com
```
Comment (matthew-robinson-ons) [inline _infra/helm/responses-dashboard/values.yaml]: To revert before merge

File: _infra/helm/responses-dashboard/templates/frontendconfig.yaml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+{{- if .Values.ingress.enabled }}
+apiVersion: networking.gke.io/v1beta1
```
Comment (LJBabbage) [inline _infra/helm/responses-dashboard/templates/frontendconfig.yaml:2]: Couldn't work out this table https://cloud.google.com/kubernetes-engine/docs/how-to/ingress-configuration#feature_comparison as nothing seems to have the superscript (B) do we need it to be v1beta1?

**Review by LJBabbage — COMMENTED** [review]
Just one query.

**Review by arroyoAle — APPROVED** [review]
Headache to test but it all works

File: _infra/helm/responses-dashboard/templates/frontendconfig.yaml (hunk 4 above)
Comment (matthew-robinson-ons) [reply _infra/helm/responses-dashboard/templates/frontendconfig.yaml:2]: This is needed i believe, There isn't a v1 version available. Also couldn't work out the table though

**Review by LJBabbage — APPROVED** [review]

**Pull request merged**